pysoa/client/client.py

```diff
@@ -364,9 +364,15 @@ def _expand_objects(self, objs_to_expand):
                                 value = exp_action_response.body[expansion_node.response_field]
                                 # Add the expansion value to the object and remove the
                                 # source field.
-                                del exp_object[expansion_node.source_field]
-                                exp_object[expansion_node.dest_field] = value
-
+                                if expansion_node.discard_source_field:
+                                    del exp_object[expansion_node.source_field]
+                                if expansion_node.discard_dest_field_id_key:
+                                    # value is a dict with one item. We want to throw away the id key,
+                                    # only put the object in place
+                                    (_, dest_obj), = value.items()
```

Comment:
dest_obj  = value.values()[0]

Reply:
hmmm
TypeError: 'dict_values' object does not support indexing

Reply:
You shouldn't use `items()` here, because it returns something different in Python 3 than it does in Python 2.

Reply:
ah... right, in Python 3 `items()` and `values()` are iterators, not lists... 
Still you can do `dest_obj, = value.values()` which which will work in both.

Reply:
Yep, that should work fine, as long as this for sure only _**ever**_ has one value. If it could possibly have more than one, you'd get `ValueError: too many values to unpack`.

Reply:
I believe this is a misunderstanding of how expansions are supposed to work.

what's supposed to happen is that you call the SOA action once per _type_, and you pass an array of all the IDs of all the objects of that type. The action returns a dict mapping those IDs to the expansion responses, and the client is supposed to re-thread the dict values onto the object.

The point of expansions is that if I have a list response of 50 events, and I expand, say, "format" on those events, the system makes a single SOA call and passes the set of _distinct_ format_ids of the events (so if the events have all the same format it's only inflating one obejct), and the SOA action responds with a dict of format_id -> format, then the server applies those format objects to the events.

The pysoa implementation here seems to be making an SOA call per object in the response, which is why the expansion action response here has a single value.